components/Feed/Caption.tsx

```diff
@@ -0,0 +1,45 @@
+"use client";
+import { useEffect, useState } from "react";
+
+export default function Caption({ caption }: { caption: string | null }) {
+  // track if the caption is truncated
+  const [isCaptionTruncated, setIsCaptionTruncated] = useState<boolean>(false);
+  // caption shown on the video slide
+  const [displayCaption, setDisplayCaption] = useState<string | null>(caption);
+  // track if the caption is expanded
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
+  const truncationCharacterLimit = 80;
```

Comment:
I usually set constants in `ALL_CAPS` to make it clear that it's a hard-coded, universal value and not something that's determined by props or state.